src/VisualStudio/TestUtilities/OutOfProcess/OutOfProcComponent.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.VisualStudio.Test.Utilities.InProcess;
+
+namespace Roslyn.VisualStudio.Test.Utilities.OutOfProcess
+{
+    /// <summary>
+    /// Base class for all components that run outside of the Visual Studio process.
```

Comment:
Is composition a better pattern here than inheritance? Is this the thing forcing the _InProc types to be public, which seems odds since that's really an implementation detail from the perspective of the calling test?